rust/src/common/config.rs

```diff
@@ -60,7 +70,7 @@ impl Configuration {
         Ok(Configuration { map })
     }
 
-    pub fn new_with_config(conf_map: HashMap<String, String>) -> io::Result<Self> {
+    pub fn new_with_config(conf_map: HashMap<String, String>) -> ConfigurationResult<Self> {
```

Comment:
I would just add `XmlParseError` to the main Error class of the crate and return a `crate::Result` here instead of a wrapper Error